File: backoffice/src/diningTable/domain/valueObjects/DiningTableNumber.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain                              import IntValueObject
from ..exceptions.InvalidDiningTableNumberException import InvalidDiningTableNumberException

"""
 *
 * Class 
 *
"""

class DiningTableNumber( IntValueObject ):

    """
     *
     * Constants 
     *
    """

    __MINIMUN_NUMBER = 1
    __MAXIMUM_NUMBER = 255

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : int ) -> None:
        super().__init__( value )
        if not self.__isValid():
            raise InvalidDiningTableNumberException( value )
    
    def __isValid( self ) -> bool:
        return self.isGreaterThanOrEqual( IntValueObject( self.__MINIMUN_NUMBER ) ) and \
            self.isLessThanOrEqual( IntValueObject( self.__MAXIMUM_NUMBER ) )

File: preparation/src/diningTable/domain/valueObjects/DiningTableNumber.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class DiningTableNumber : IntValueObject {

        /* 
         * 
         * Constants
         *
        */

        private const int MINIMUN_NUMBER = 1;
        private const int MAXIMUM_NUMBER = 255;

        /* 
         * 
         * Methods
         *
        */

        public DiningTableNumber( int value ) : base( value ) {
            if( !isValid() ) {
                throw new InvalidDiningTableNumberException( value );
            }
        }

        private bool isValid() {
            return isGreaterThanOrEqual( new IntValueObject( MINIMUN_NUMBER ) ) &&
                isLessThanOrEqual( new IntValueObject( MAXIMUM_NUMBER ) );
        }

    }

}